maestro-cli/src/main/java/maestro/cli/command/UploadCommand.kt

```diff
@@ -0,0 +1,191 @@
+        return 0
+    }
+
+    private fun getCachedAuthToken(): String? {
+        if (!cachedAuthTokenFile.exists()) return null
+        if (cachedAuthTokenFile.isDirectory()) return null
+        val cachedAuthToken = cachedAuthTokenFile.readText()
+        return if (client.isAuthTokenValid(cachedAuthToken)) {
+            cachedAuthToken
+        } else {
+            message("Existing auth token is invalid or expired")
+            cachedAuthTokenFile.deleteIfExists()
+            null
+        }
+    }
+
+    private fun setCachedAuthToken(token: String?) {
+        cachedAuthTokenFile.parent.createDirectories()
+        if (token == null) {
+            cachedAuthTokenFile.deleteIfExists()
+        } else {
+            cachedAuthTokenFile.writeText(token)
+        }
+    }
+
+    private fun triggerSignInFlow(): String {
+        message("No auth token found")
+        val email = prompt("Sign In or Sign Up using your company email address:")
+        var isLogin = true
+        val requestToken = client.magicLinkLogin(email).getOrElse { loginError ->
+            if (loginError.code == 403 && loginError.body?.string()?.contains("not an authorized email address") == true) {
+                isLogin = false
+                message("No existing team found for this email domain")
+                val team = prompt("Enter a team name to create your team:")
+                client.magicLinkSignUp(email, team).getOrElse { signUpError ->
+                    throw CliError(signUpError.body?.string() ?: signUpError.message)
+                }
+            } else {
+                throw CliError(loginError.body?.string() ?: loginError.message)
+            }
+        }
+
+        if (isLogin) {
+            message("We sent a login link to $email. Click on the link there to finish logging in...")
+        } else {
+            message("We sent an email to $email. Click on the link there to finish creating your account...")
+        }
+
+        while (true) {
+            val errResponse = when (val result = client.magicLinkGetToken(requestToken)) {
+                is Ok -> {
+                    if (isLogin) {
+                        message("✅ Login successful")
+                    } else {
+                        message("✅ Team created successfully")
+                    }
+                    setCachedAuthToken(result.value)
+                    return result.value
+                }
+                is Err -> result.error
+            }
+            val errorMessage = errResponse.body?.string() ?: errResponse.message
+            if (
+                "Login process not complete" !in errorMessage
+                && "Email is not authorized" !in errorMessage
+            ) {
+                throw CliError("Failed to get auth token (${errResponse.code}): $errorMessage")
+            }
+            Thread.sleep(1000)
+        }
+    }
+
+    companion object {
+
+        init {
+            AnsiConsole.systemInstall()
+        }
```

Comment:
Curious, can this happen in the parent command or should it happen in every subcommand?

Reply:
Good suggestion! It can probably be moved to App.kt. Updated!